### PR TITLE
Improve Language Library's OFT specification items

### PR DESCRIPTION
**Pull request by sophokles73**
THis is for #197]

Some of the items have been dropped because they should be included in
the specification documents of uProtocol's basic types instead.

A title has been added to all spec items which indicates the item's
purpose.

Some items have been split up into multiple items and/or rephrased to
improve readability.

### CODE REVIEW

File: languages.adoc [hunk 1]
```diff
@@ -20,76 +20,86 @@ SPDX-License-Identifier: Apache-2.0
...
 --
-* *MUST* implement serializers, validators, and builders for the link:../basics/README.adoc[uProtocol data types] using the link:../up-core-api/README.adoc[up-core-api data model].
+* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
 --
 
-[.specitem,oft-sid="req~up-language-transport-api~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="req~up-language-comm-api~1",oft-needs="impl",oft-title="Library declares Communication Layer API"]
 --
-* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
+* *MUST* declare the language specific link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) API], such that the interface can be implemented by the transport libraries.
```
Comment (stevenhartley) [inline languages.adoc:39]: Transport libraries do not implement L2 APIs

File: languages.adoc [hunk 2]
```diff
@@ -20,76 +20,86 @@ SPDX-License-Identifier: Apache-2.0
...
 
-[.specitem,oft-sid="req~up-language-structure~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="dsn~up-language-structure~1",oft-title="Library uses uProtocol namespace"]
 --
-* The top level namespace (package name) *MUST* contain `uprotocol` and *SHOULD* contain `org.eclipse.uprotocol` 
+* *MUST* use the programming language specific namespace concept, if availabe, to indicate that the library belongs to the _uProtocol_ project. If the language supports hierarchical namespaces (like Java does), then `org.eclipse.uprotocol` *SHOULD* be used. Otherwise, a namespace of `uprotocol` *MUST* be used*.
 --
 
-[.specitem,oft-sid="req~up-language-documentation~1",oft-needs="impl,utest"]
+[.specitem,oft-sid="req~up-language-documentation~1",oft-needs="uman",oft-title="Library contains adequate README file"]
```
Comment (stevenhartley) [inline languages.adoc:57]: uman is a new one but I gather it refers to user manual

**Review by stevenhartley — CHANGES_REQUESTED** [review]
minor comment

File: languages.adoc (hunk 2 above)
Comment (sophokles73) [reply languages.adoc:57]: it does

File: languages.adoc (hunk 1 above)
Comment (sophokles73) [reply languages.adoc:39]: FMPOV they can indeed. Why shouldn't they? If there is a better, transport specific way to implement the RpcClient (see zenoh), then why not?

Comment (sophokles73) [reply languages.adoc:39]: @stevenhartley any further thoughts?

Comment (stevenhartley) [reply languages.adoc:39]:
```suggestion
* *MUST* declare the language specific link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) API], such that the interface *MAY* be implemented by the transport libraries.
```

**Comment by sophokles73** [comment]
@stevenhartley can you take another look?

**Review by stevenhartley — APPROVED** [review]
LGTM

**Pull request merged**